include/libdnf/base/goal.hpp

```diff
@@ -54,53 +54,285 @@ class Goal {
     explicit Goal(const libdnf::BaseWeakPtr & base);
     explicit Goal(libdnf::Base & base);
     ~Goal();
+
+    // TODO(jmracek) Implemented not yet
     void add_module_enable(const std::string & spec);
-    /// Prevent reinstallation by adding of already installed packages with the same NEVRA
+
+    /// Add install job to the goal for future resolvement. The operation will not result in package reinstall when a
```

Comment:
_"Add install job"_ - this can be addressed later but I think this lacks the definition of what a "job" really is. We either shouldn't use this word in libdnf context or we should have some sort of definition of what it actually means.

_"future resolvement"_ - as @dmach said, please use "dependency resolution" or just "resolution", "resolvement" seems to be a word according to dictionary, but I've never head it in this context. I'd also prefer not to use "future" here, as it feels to me like it's a vague indication of something that is far off.

What about something like:
> Add the `spec` specifier for installation in the goal.